File: tests/rosetta/transpiler/php/case-sensitivity-of-identifiers.php
<?php
ini_set('memory_limit', '-1');
function _len($x) {
    if (is_array($x)) { return count($x); }
    if (is_string($x)) { return strlen($x); }
    return strlen(strval($x));
}
function _str($x) {
    if (is_array($x)) {
        $isList = array_keys($x) === range(0, count($x) - 1);
        if ($isList) {
            $parts = [];
            foreach ($x as $v) { $parts[] = _str($v); }
            return '[' . implode(' ', $parts) . ']';
        }
        $parts = [];
        foreach ($x as $k => $v) { $parts[] = _str($k) . ':' . _str($v); }
        return 'map[' . implode(' ', $parts) . ']';
    }
    if (is_bool($x)) return $x ? 'true' : 'false';
    if ($x === null) return 'null';
    return strval($x);
}
function main() {
  $pkg_dog = 'Salt';
  $Dog = 'Pepper';
  $pkg_DOG = 'Mustard';
  $packageSees = function($d1, $d2, $d3) use (&$packageSees, $pkg_dog, $Dog, $pkg_DOG) {
  echo rtrim('Package sees: ' . $d1 . ' ' . $d2 . ' ' . $d3), PHP_EOL;
  return ['pkg_dog' => true, 'Dog' => true, 'pkg_DOG' => true];
};
  $d = $packageSees($pkg_dog, $Dog, $pkg_DOG);
  echo rtrim('There are ' . _str(_len($d)) . ' dogs.
'), PHP_EOL;
  $dog = 'Benjamin';
  $d = $packageSees($pkg_dog, $Dog, $pkg_DOG);
  echo rtrim('Main sees:   ' . $dog . ' ' . $Dog . ' ' . $pkg_DOG), PHP_EOL;
  $d['dog'] = true;
  $d['Dog'] = true;
  $d['pkg_DOG'] = true;
  echo rtrim('There are ' . _str(_len($d)) . ' dogs.
'), PHP_EOL;
  $Dog = 'Samba';
  $d = $packageSees($pkg_dog, $Dog, $pkg_DOG);
  echo rtrim('Main sees:   ' . $dog . ' ' . $Dog . ' ' . $pkg_DOG), PHP_EOL;
  $d['dog'] = true;
  $d['Dog'] = true;
  $d['pkg_DOG'] = true;
  echo rtrim('There are ' . _str(_len($d)) . ' dogs.
'), PHP_EOL;
  $DOG = 'Bernie';
  $d = $packageSees($pkg_dog, $Dog, $pkg_DOG);
  echo rtrim('Main sees:   ' . $dog . ' ' . $Dog . ' ' . $DOG), PHP_EOL;
  $d['dog'] = true;
  $d['Dog'] = true;
  $d['pkg_DOG'] = true;
  $d['DOG'] = true;
  echo rtrim('There are ' . _str(_len($d)) . ' dogs.'), PHP_EOL;
}
main();

File: tests/rosetta/transpiler/Clojure/case-sensitivity-of-identifiers.clj
(ns main (:refer-clojure :exclude [main]))

(require 'clojure.set)

(defn in [x coll]
  (cond (string? coll) (clojure.string/includes? coll x) (map? coll) (contains? coll x) (sequential? coll) (some (fn [e] (= e x)) coll) :else false))

(defn padStart [s w p]
  (loop [out (str s)] (if (< (count out) w) (recur (str p out)) out)))

(def nowSeed (atom (let [s (System/getenv "MOCHI_NOW_SEED")] (if (and s (not (= s ""))) (Integer/parseInt s) 0))))

(declare main)

(declare DOG d dog main_Dog main_pkg_DOG main_pkg_dog)

(defn packageSees [packageSees_d1 packageSees_d2 packageSees_d3]
  (try (do (println (str (str (str (str (str "Package sees: " packageSees_d1) " ") packageSees_d2) " ") packageSees_d3)) (throw (ex-info "return" {:v {"pkg_dog" true "Dog" true "pkg_DOG" true}}))) (catch clojure.lang.ExceptionInfo e (if (= (ex-message e) "return") (get (ex-data e) :v) (throw e)))))

(defn main []
  (try (do (def main_pkg_dog "Salt") (def main_Dog "Pepper") (def main_pkg_DOG "Mustard") (def d (packageSees pkg_dog Dog pkg_DOG)) (println (str (str "There are " (str (count d))) " dogs.\n")) (def dog "Benjamin") (def d (packageSees pkg_dog Dog pkg_DOG)) (println (str (str (str (str (str "Main sees:   " dog) " ") Dog) " ") pkg_DOG)) (def d (assoc d "dog" true)) (def d (assoc d "Dog" true)) (def d (assoc d "pkg_DOG" true)) (println (str (str "There are " (str (count d))) " dogs.\n")) (def Dog "Samba") (def d (packageSees pkg_dog Dog pkg_DOG)) (println (str (str (str (str (str "Main sees:   " dog) " ") Dog) " ") pkg_DOG)) (def d (assoc d "dog" true)) (def d (assoc d "Dog" true)) (def d (assoc d "pkg_DOG" true)) (println (str (str "There are " (str (count d))) " dogs.\n")) (def DOG "Bernie") (def d (packageSees pkg_dog Dog pkg_DOG)) (println (str (str (str (str (str "Main sees:   " dog) " ") Dog) " ") DOG)) (def d (assoc d "dog" true)) (def d (assoc d "Dog" true)) (def d (assoc d "pkg_DOG" true)) (def d (assoc d "DOG" true)) (println (str (str "There are " (str (count d))) " dogs."))) (catch clojure.lang.ExceptionInfo e (if (= (ex-message e) "return") (get (ex-data e) :v) (throw e)))))

(defn -main []
  (main))

(-main)
